### PR TITLE
SE: Add repro from ITs

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Related to #6070

Accepts it as TP. The `??` is visited because of the type mismatch, or because the original value was null.

>as check with null-coalescing sets misleading null constraint in the Null case => graph as Flow<TIn, TOut, TMat> ?? new Flow<TIn, TOut, TMat>(graph.Module); sources\akka.net\src\core\Akka.Streams\Dsl\Flow.cs

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs [hunk 1]
```diff
@@ -22,6 +25,11 @@ public void NullCoalescenceOperator_Null_Noncompliant2()
             name = name ?? name.ToString(); // Noncompliant
         }
 
+        public void NullCoalesce_Conversion(Exception arg)  // Classes doesn't make sense here. They just demonstrate the flow.
+        {
+            var value = arg as IDisposable ?? new FileStream(arg.Message, FileMode.Open);  // Noncompliant, if arg is null, it will throw
+        }
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs:31]: I think it is more nuanced than that.
```suggestion

        public void NullCoalesce_Conversion_DownCast(ArgumentException arg)        {
            var value = arg as Exception ?? new FileStream(arg.Message, FileMode.Open);  // Noncompliant, arg must be null on the right side
        }

        public void NullCoalesce_Conversion_UpCast(Exception arg)        {
            var value = arg as ArgumentException ?? new FileStream(arg.Message, FileMode.Open);  // Compliant, arg can be null or another Exception type on the right side
        }        
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs [hunk 2]
```diff
@@ -22,6 +25,11 @@ public void NullCoalescenceOperator_Null_Noncompliant2()
             name = name ?? name.ToString(); // Noncompliant
         }
 
+        public void NullCoalesce_Conversion(Exception arg)  // Classes doesn't make sense here. They just demonstrate the flow.
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs:28]: I don't understand the comment. Is it needed to understand the test.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs:31]: The 2nd says `Compliant. I guess it was a typo, as there could be null

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs:28]: It's gone after the changes you've proposed. It was just weird to have `value` as disposable exception OR file stream created from the exception message

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs]: We don't know for sure that it is null. It likely isn't. So I would call it compliant. But that is debatable.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs [hunk 3]
```diff
@@ -22,6 +25,15 @@ public void NullCoalescenceOperator_Null_Noncompliant2()
...
         }
 
+        public void NullCoalesce_Conversion_DownCast(ArgumentException arg)
+        {
+            var value = arg as Exception ?? new Exception(arg.Message);     // Noncompliant, arg must be null on the right side
+        }
+        public void NullCoalesce_Conversion_UpCast(Exception arg)
+        {
+            var value = arg as ArgumentException ?? new ArgumentException(arg.Message);     // Noncompliant, arg can be null or another Exception type on the right side
+        }
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs:35]:
```suggestion
        public void NullCoalesce_Conversion_DownCast(AggregateException arg)
        {
            var value = arg as Exception ?? new Exception(arg.Message);     // Noncompliant, arg must be null on the right side
        }

        public void NullCoalesce_Conversion_UpCast(Exception arg)
        {
            var value = arg as AggregateException ?? new AggregateException(arg.Message);     // Noncompliant, arg can be null or another Exception type on the right side
        }
```

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. Minor: Using `AggregateException` makes the code a bit more reasonable. Feel free to use it or not. I also added a new line between the two methods.

**Pull request merged**